qa/rpc-tests/test_framework/mininode.py

```diff
@@ -719,16 +968,39 @@ def is_valid(self):
         return True
 
     def solve(self):
+        if self.aux_pow:
+            self.rehashaux()
+            aux_target1 = uint256_from_compact(self.aux_pow.commit_bits)
+            aux_target2 = 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
+            bias = self.get_bias()
+            aux_target1 = aux_target1 << bias
+            assert (aux_target1 <= aux_target2)
+            aux_target2 = aux_target2 >> bias
+            while (self.aux_hash1 > aux_target1) or (self.aux_hash2 > aux_target2):
+                self.aux_pow.aux_nonce += 1
+                self.rehashaux()
+            # First remove the existing commitment, if there is one.  There isn't a way
+            # to identify the commitment, so we presume that one is present if the
+            # coinbase miner string is large enough to contain one.
+            if len(self.vtx[0].vin[0].scriptSig) > 32:
```

Comment:
Is this guaranteed to *never* be only the commitment? If not, this should be `>= 32`.

Reply:
Good catch. I should probably make it `>= 32` to be easier to read, but the answer is actually "it's complicated." The coinbase string also needs the BIP 34 height-in-coinbase, which adds a few bytes, which is why I think I originally did `> 32`. But BIP 34 is going away with the protocol cleanup, after which the auxpow commitment is the only required field.